src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -211,9 +218,57 @@ private IEnumerable<IConfigurationSection> GetDotnetFeatureDefinitionSections()
 
         private IEnumerable<IConfigurationSection> GetMicrosoftFeatureDefinitionSections()
         {
-            return _configuration.GetSection(MicrosoftFeatureManagementFields.FeatureManagementSectionName)
-                .GetSection(MicrosoftFeatureManagementFields.FeatureFlagsSectionName)
-                .GetChildren();
+            var featureDefinitionSections = new List<IConfigurationSection>();
+
+            FindFeatureFlags(_configuration, featureDefinitionSections);
+
+            return featureDefinitionSections;
+        }
+
+        private void FindFeatureFlags(IConfiguration configuration, List<IConfigurationSection> featureDefinitionSections)
+        {
+            if (!(configuration is IConfigurationRoot configurationRoot) ||
+                configurationRoot.Providers.Any(provider =>
+                    !(provider is ConfigurationProvider) && !(provider is ChainedConfigurationProvider)))
```

Comment:
Given there is recursion being used for ChainedConfigurationProvider, don't we need to recursively validate first?